web-client/src/presenter/computeds/AdvancedSearch/practitionerSearchHelper.ts

```diff
@@ -29,9 +29,10 @@ type PractitionerSearchHelperResult = {
   showStateSelect?: boolean;
   activePage?: number;
   formattedSearchResults?: FormattedPractitionerSearchResultType[];
-  numberOfResults?: number;
+  numberOfResults?: string;
```

Comment:
Is there a way that these can be required instead of optional? It makes sense to me that if no results were returned `numberOfResults` would be "0" or `showPaginator` would be "false"

Reply:
It's a possibility. Here's the thought with this:
- If there are results, an entire `PractitionerSearchHelperResult` will be returned
- If there are no results, it will only return `showNoMatches` and `showSearchResults`, and "only return what it needs"
- If the search hasn't been run yet, it will omit `showNoMatches` and `showSearchResults`, as well as everything else. the absence of these two props is how we know the search hasn't been run yet.

I will concede it's an iffy way of keeping track of these three states, and I'm open to suggestions on making this particular thing better. Should we use separate types for the different returns?

Reply:
Never mind, it just works. let me know what you think.